File: Codepath/Week4-Review/java/createDictionary.java
/*
 * Problem: Design a method to find the number of occurrences of any given word in a book.
 * A word is represented as a String and a book is represented as a list of strings.
 * This method will be called multiple times.
*/
HashMap<String, Integer> createDictionary(String[] book) {
        HashMap<String, Integer> dictionary = new HashMap<>();
        for (String word : book) {
            word = word.trim();
            if (!word.equals("")) {
                word = word.toLowerCase();
                if (!dictionary.containsKey(word)) {
                    dictionary.put(word, 1);
                }
                dictionary.put(word, dictionary.get(word) + 1);
            }
        }
        return dictionary;
    }

    int getFrequency(String[] book, String word) {
        HashMap<String, Integer> dictionary = createDictionary(book);
        if (dictionary == null || word == null) return -1;
        word = word.toLowerCase();
        if (dictionary.containsKey(word)) {
            return dictionary.get(word);
        }
        return 0;
    }

File: Codepath/Week4-Review/python/createDictionary.py
# Problem: Design a method to find the number of occurrences of any given word in a book. 
# A word is represented as a String and a book is represented as a list of strings.
# This method will be called multiple times.
def createDictionary(book):
    if not book:
       return None
    dictionary = {}
    for word in book:
        word = word.strip()
        if word:
            word = word.lower()
            if word not in dictionary:
                dictionary[word] = 1
            else:
                dictionary[word] = dictionary[word] + 1
    return dictionary

def getFrequency(book, word):
    dictionary = createDictionary(book)
    if not book or not word:
        return -1
    word = word.strip().lower()
    if word in dictionary:
        return dictionary[word]
    else:
        return 0
